File: ext/ffmpeg.c
/* Copyright (c)2008 Antonin Amand.
 * Licensed under the Ruby License. See LICENSE for details.
 *
 */
#include "ffmpeg.h"

VALUE rb_mFFMPEG;

void
Init_FFMPEG_core()
{
    //fprintf(stderr, "----------------- init FFMPEG ------------------\n");
    // initialize ffmpeg
    
    av_register_all();
    avcodec_register_all();

    // initialize Module and Class
    rb_mFFMPEG = rb_define_module("FFMPEG");
    rb_define_const(rb_mFFMPEG, "LIBAVCODEC_VERSION", LONG2NUM((long)avcodec_version()));
    
    Init_FFMPEGFormat();
    Init_FFMPEGInputFormat();
    //Init_FFMPEGOutputFormat();
    Init_FFMPEGStream();
    Init_FFMPEGCodec();
    Init_FFMPEGFrame();
}

File: ext/ffmpeg.rb
# Copyright (c)2008 Antonin Amand.
# Licensed under the Ruby License. See LICENSE for details.
#
begin
  require 'HFL_core'
rescue LoadError
  require File.dirname(__FILE__) + '/FFMPEG_core'
end

module FFMPEG
  
  class InputFormat
    
    def first_video_stream
      streams.find { |st| st.type == :video }
    end
    
    def first_audio_stream
      streams.find { |st| st.type == :audio }
    end
    
  end
  
  class Stream
    
    def type
      codec.type
    end
    
  end
  
end